packages/transformer/src/test/standalone/IModelTransformerHub.test.ts

```diff
@@ -609,19 +609,13 @@ describe("IModelTransformerHub", () => {
             const aspects =
               [...branch.db.queryEntityIds({ from: "BisCore.ExternalSourceAspect" })]
               .map((aspectId) => branch.db.elements.getAspect(aspectId).toJSON()) as ExternalSourceAspectProps[];
+              // FIXME: It seems as if deep subsetEqual requires that the array of elements be in the same order between expected and actual.
```

Comment:
I think we provide the `.deep.subsetEqual` implementation as a custom matcher. We can probably support what you want by rewriting it to compare arrays using `.includes`. Or since that's n^2 it could load one into `Set` and do a linear `.has` check

Reply:
I'm looking at this some more and calling Object.keys on an array gives the keys as indices (makes sense). Thats why they have to be in the same order currently. Do your suggestions still work in this case? Since advancedDeepEqual is recursive (basically breaking down each object until its just a simple key value pair, it feels strange to somehow determine which index I'm supposed to be breaking down for comparison when the expected and actual aren't in the same order. I'm fine to leave it as is and maybe call that a TODO: instead of a fixme? 
Code snippet for more context:
```
return (eSize === aSize || !!options.useSubsetEquality) && Object.keys(e).every(
        (keyOfE) =>
          (keyOfE in a || options.considerNonExistingAndUndefinedEqual) &&
          normalizedClassNameProps.includes(keyOfE)
            ? advancedDeepEqual(normalizeClassName(e[keyOfE]), normalizeClassName(a[keyOfE]))
            : advancedDeepEqual(e[keyOfE], a[keyOfE], options)
      );
```

Reply:
my suggestion works if you add a check for `Array.isArray` to handle arrays specially from other values where `typeof x === "object"`

Reply:
Am I mistaken that your suggestion only works for simple elements in an array? Includes with js objects (which is the case I discovered this issue in) will check that the two objects are exactly the same (same identity?)

Reply:
discussed this in call